dask/dataframe/shuffle.py

```diff
@@ -228,6 +228,11 @@ def set_index(
             df, index2, repartition, npartitions, upsample, partition_size
         )
 
+        if pd.isna(mins + maxes).all():
```

Comment:
With pyarrow strings, we may get `[pd.NA, pd.NA...]` in `mins` and `maxes`, and `[None, None...]` in `divisions, which makes them unsortable - an exception is thrown on calling `sorted`.